### DISCUSSION TITLE
当模型推理sql，遇到复杂字段（难以理解的字段）时，我们应该做些什么？

### DISCUSSION THREAD

**Discussion by AOnlineFish** [eosphoros-ai/DB-GPT-Hub#156] [Q&A]
# 讨论：处理SQL模型推理中的复杂字段

## 问题描述

在使用机器学习模型进行SQL推理时，我们经常遇到难以理解（或者本身含义并非实际含义）的复杂字段，[如图1](https://github.com/eosphoros-ai/DB-GPT-Hub/assets/148038819/97c1eb84-002c-4f81-9b5f-d0e3653c096f)。这些字段可能定义不明确、关系复杂或格式非标准，给模型的解释和预测准确性带来了重大挑战。

## 寻求解决方案

我们的目标是探索有效处理这些复杂字段的策略。这可能包括数据预处理、特征工程、模型选择或任何可以简化复杂性以提高模型性能的创新方法。

## 邀请社区参与

这是一个更广泛讨论的起点。我邀请社区分享见解、经验和潜在的解决方案来应对这一挑战。让我们合作，找到处理SQL模型推理中复杂字段的有效方法。

![微信图片_20231201202226](https://github.com/eosphoros-ai/DB-GPT-Hub/assets/148038819/97c1eb84-002c-4f81-9b5f-d0e3653c096f)

**Comment by qidanrui** [comment]
非常好的讨论帖哈~
我先提一种可能性，在我们的sql_data_process.py里面，有给模型创造prompt的语句：
![Screen Shot 2023-12-01 at 8 33 27 PM](https://github.com/eosphoros-ai/DB-GPT-Hub/assets/10005367/34af454c-5dd1-4a27-a19c-81526ba5c2c1)
或许在这些table的名字后面，可以加每个列名对应的语义？这样在训练中，相当于让模型学到了每个列名的语义。

**Reply by AOnlineFish** [reply]
我觉得可以尝试一下~

**Comment by AOnlineFish** [comment]
@yihong0618 @csunny @wangzaistone 请问各位老师，对这个复杂字段有什么见解嘛？
